packages/flutter_tools/lib/src/flx.dart

```diff
@@ -77,17 +78,27 @@ Future<Null> build({
   if (!precompiledSnapshot && previewDart2) {
     ensureDirectoryExists(applicationKernelFilePath);
 
-    final String kernelBinaryFilename = await compile(
-      sdkRoot: artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
-      incrementalCompilerByteStorePath: fs.path.absolute(getIncrementalCompilerByteStoreDirectory()),
-      mainPath: fs.file(mainPath).absolute.path,
-      outputFilePath: applicationKernelFilePath,
-      depFilePath: depfilePath,
-      trackWidgetCreation: trackWidgetCreation,
-      fileSystemRoots: fileSystemRoots,
-      fileSystemScheme: fileSystemScheme,
-      packagesPath: packagesPath,
-    );
+    String kernelBinaryFilename;
+    if (residentCompiler != null) {
```

Comment:
Sorry, I'm not getting why passing `residentCompiler` into this `build` method indicates we are doing recompile. Who is going to pass non-null `residentCompiler` into `build`?

Reply:
The idea is that there are devices which build at `startApp` - `IOSSimulator` and `FlutterTesterDevice`.  And then there is also `ResidentCompiler generator` in `FlutterDevice`, which is used for hot reloads. So, I thought it would be beneficial to start one instance of `ResidentCompiler` and use it both for initial compilation during `flx.build` and for subsequence hot reloads.

I'm not particularly attached to this idea, I can just do in `FlutterTesterDevice` what `IOSSimulator` - run one-off compilation.